src/rviz_visual_tools.cpp

```diff
@@ -334,7 +336,7 @@ bool RvizVisualTools::waitForSubscriber(const ros::Publisher& pub, double wait_t
                                                   "will be lost.");
       return false;
     }
-    ros::spinOnce();
+    vis_marker_queue_.callAvailable(ros::WallDuration());
```

Comment:
Well the code tries to block until there is a subscriber so deleting it would result in a different behavior.
But I just bumped into this again when using `ros::spin()` and there PR didn't help so I think it would be better to drop this or provide a non blocking interface. I can come up with a patch if more people agree.

Reply:
I added this function because often important stuff was published to Rviz before Rviz was done loading, and the visualizations would not work correctly. It was great for demo / testing code, not sure its useful in a production env though. Maybe just warn API users not to use this function in production systems

Reply:
but is there a way to make it work without calling this function? I guess one could take the `markers_` and do the publish manually but that sounds wrong, or set `pub_rviz_markers_connected_` and `pub_rviz_markers_waited_` to true..
Maybe we could add a variable to the constructor to disable the connection check or have a `trigger` variant to not do it. Honestly I think the check should be disabled by default. What do you think?

Reply:
I just verified in a small C++ test script that spinning is fully independent of node connection handling and I would have been surprised by anything else.
So even if you don't spin, `getNumSubscribers` reports new subscribers as well.

@jspricke From all that I can see the change in behavior you refer to actually means fixing the bug.
The method is not supposed to spin *any* queue. It's supposed to wait and poll*.

-- Slightly off-topic

* - I was convinced there is an API somewhere to get notified about new subscribes/unsubscribes and that would be even better. But I can't seem to find it